File: testing-rb/lib/homophoner.rb
class Homophoner
  attr_reader :dictionary

  def initialize(homophones)
    @dictionary = Hash.new
    homophones.each do |wordlist|
      wordlist.each do |word|
        @dictionary[word] = (wordlist - [word]).sort
      end
    end
  end

  def lookup(word)
    dictionary.fetch(word, [])
  end

  def to_s
    "<" + self.class.name + ": " + dictionary.to_s + ">"
  end
end

File: testing-jasmine/lib/homophoner.js
var _ = require('lodash-node');

module.exports = Homophoner;

function Homophoner(homophones) {
  this.dictionary = {};

  homophones.map((wordlist) => {
    wordlist.map((word) => {
      this.dictionary[word] = _.sortBy(_.difference(wordlist, [word]));
    });
  });

  this.lookup = function(word) {
    return this.dictionary[word] || [];
  }
}
